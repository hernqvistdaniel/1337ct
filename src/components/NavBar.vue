<script lang="ts">
export default {
  props: ['view'],
  location: String,
  methods: {
    handleFilter(filter) {
      this.$emit('handleFilter', filter)
    },
    handleSort(sort) {
      this.$emit('handleSort', sort)
    },
    handleView(view) {
      this.$emit('handleView', view)
    }
  }
}
</script>

<template>
  <div class="navbar">
    <button v-if="view !== 'list'" @click="handleView($event.target.value)" value="list" class="navitem">View as
      list</button>
    <button v-if="view == 'list'" @click="handleView($event.target.value)" value="cards" class="navitem">View as
      cards</button>
    <button @click="handleSort($event.target.value)" value="phone" class="navitem">Sort on phonenumber?</button>
    <p>Office:</p>
    <label class="navitem">
      <input name="officeSelection" v-model="location" type="text">
      <button @click="handleFilter(location)">s√∂k</button>
    </label>
    <label class="navitem">
      All
      <input name="officeSelection" @click="handleFilter($event.target.value)" value="all" type="radio">
    </label>
  </div>
</template>

<style scoped>
.navbar {
  width: 100%;
  height: 50px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  text-align: center;
  margin: 10px;
  display: flex;
  justify-content: end;
  flex-direction: row;
  align-items: center;
}

.navitem {
  margin: 10px;
}
</style>
