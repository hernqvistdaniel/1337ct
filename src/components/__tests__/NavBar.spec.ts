import { describe, it, expect } from 'vitest'

import { shallowMount } from '@vue/test-utils'
import NavBar from '../NavBar.vue'

describe('GalleryCard.vue', () => {
  const wrapper = shallowMount(NavBar, { props: {} })

  it('renders properly', () => {
    expect(wrapper.html()).toBeTruthy()
  })

  it('matches a snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})