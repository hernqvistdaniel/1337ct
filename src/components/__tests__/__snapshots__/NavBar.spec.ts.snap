// Vitest Snapshot v1

exports[`GalleryCard.vue > matches a snapshot 1`] = `
"<div class=\\"navbar\\" data-v-4295d220=\\"\\"><button class=\\"navitem\\" value=\\"list\\" data-v-4295d220=\\"\\">View as list</button>
  <!--v-if--><button class=\\"navitem\\" value=\\"phone\\" data-v-4295d220=\\"\\">Sort on phonenumber?</button>
  <p data-v-4295d220=\\"\\">Office:</p><label class=\\"navitem\\" data-v-4295d220=\\"\\"><input name=\\"officeSelection\\" type=\\"text\\" data-v-4295d220=\\"\\"><button data-v-4295d220=\\"\\">s√∂k</button></label><label class=\\"navitem\\" data-v-4295d220=\\"\\"> All <input name=\\"officeSelection\\" type=\\"radio\\" value=\\"all\\" data-v-4295d220=\\"\\"></label>
</div>"
`;
